cmd/ceremony/main.go

```diff
@@ -417,50 +506,73 @@ func openSigner(cfg PKCS11SigningConfig, pubKey crypto.PublicKey) (crypto.Signer
 	if err != nil {
 		return nil, nil, fmt.Errorf("failed to retrieve private key handle: %s", err)
 	}
-	if !equalPubKeys(signer.Public(), pubKey) {
-		return nil, nil, fmt.Errorf("signer pubkey did not match issuer pubkey")
+	ok, err := publicKeysEqual(signer.Public(), pubKey)
+	if !ok {
+		return nil, nil, err
 	}
-	log.Println("Retrieved private key handle")
+
 	return signer, newRandReader(session), nil
 }
 
-func signAndWriteCert(tbs, issuer *x509.Certificate, subjectPubKey crypto.PublicKey, signer crypto.Signer, certPath string, skipLints []string) error {
-	err := linter.Check(tbs, subjectPubKey, issuer, signer, skipLints)
-	if err != nil {
-		return fmt.Errorf("certificate failed pre-issuance lint: %w", err)
-	}
+func signAndWriteCert(tbs, issuer *x509.Certificate, subjectPubKey crypto.PublicKey, signer crypto.Signer, certPath string) (*x509.Certificate, error) {
```

Comment:
Document that this MUST only be called after linting has been done. Or come up with some way to enforce that constraint.

Reply:
I came up with a solution in slack and then we refined it. Those changes and a test have been committed.